Make.inc

```diff
@@ -1384,9 +1392,11 @@ OSLIBS += -lelf -lkvm -lrt -lpthread -latomic
 # make it loaded first to
 # prevent from linking to outdated system libs.
 # See #21788
+ifeq (,$(findstring aarch64,$(ARCH)))
 OSLIBS += -lgcc_s
+endif
```

Comment:
I'm unsure whether this is actually specific to AArch64 or if it's due to differences when building with `USE_BINARYBUILDER=0`. My FreeBSD x86-64 machine is too old to complete a `USE_BINARYBUILDER=0` build this century otherwise I'd just test that.

Reply:
@giordano might know?

Reply:
I don't have an answer, but I seem to remember FreeBSD uses its own fake libgcc_s for a few things, but I don't know if that's x86_64 specific or what. Anyways, this change is relevant only for the aarch64 freebsd builds, so it should be good for the time being, if someone brave wants to investigate, that can be done in a follow up PR.

Reply:
A `libgcc_s` that declares itself as GCC 4.6 (or something like that) is present on the system regardless of architecture. I had to make this change because it was getting linked in too early and causing symbol versioning issues on AArch64. Thinking about it more, I wonder if linking in `libgcc_s` earlier is necessary when using BinaryBuilder to ensure the `libgcc_s` we vendor is picked up first, and that without BinaryBuilder you don't get the correctly versioned, vendored one, so you want to delay linking in `libgcc_s`. If that's true, this part will need to be changed when we have FreeBSD AArch64 support in BinaryBuider, though that can happen alongside other necessary changes, e.g. updating the JLL builds.

Reply:
Perhaps a line or two can be added to the comment to explain this, so the next poor person trying to make sense of the Makefile has a slightly easier time